server/src/main/java/org/elasticsearch/cluster/InternalClusterInfoService.java

```diff
@@ -191,76 +196,93 @@ private void fetchIndicesStats() {
             indicesStatsRequest.store(true);
             indicesStatsRequest.indicesOptions(IndicesOptions.STRICT_EXPAND_OPEN_CLOSED_HIDDEN);
             indicesStatsRequest.timeout(fetchTimeout);
-            client.admin().indices().stats(indicesStatsRequest, ActionListener.runAfter(new ActionListener<>() {
-                @Override
-                public void onResponse(IndicesStatsResponse indicesStatsResponse) {
-                    logger.trace("received indices stats response");
-
-                    if (indicesStatsResponse.getShardFailures().length > 0) {
-                        final Set<String> failedNodeIds = new HashSet<>();
-                        for (final DefaultShardOperationFailedException shardFailure : indicesStatsResponse.getShardFailures()) {
-                            if (shardFailure.getCause()instanceof final FailedNodeException failedNodeException) {
-                                if (failedNodeIds.add(failedNodeException.nodeId())) {
-                                    logger.warn(
-                                        () -> format("failed to retrieve shard stats from node [%s]", failedNodeException.nodeId()),
-                                        failedNodeException.getCause()
-                                    );
+            client.admin()
+                .indices()
+                .stats(
+                    indicesStatsRequest,
+                    new ThreadedActionListener<>(
```

Comment:
Couldn't really find a nice way of doing this without a big changeset from the indents, no changes to the code here other than the deduplication though.

Reply:
... and the fact that we now fork to `MANAGEMENT`? Presumably that's because the deduplication is too expensive to run on the transport thread right?

Reply:
Yea, it's not a massive slowdown here from this change (maybe like 30%) but even before this change running this on a transport thread wasn't ideal for a 50k shards cluster so I didn't want to make that situation worse and changed it in this PR.